app/workers/add_facebook_friends_worker.rb

```diff
@@ -0,0 +1,19 @@
+class AddFacebookFriendsWorker
+  include Sidekiq::Worker
+
+  def perform(user_id)
+    user = User.find(user_id)
+    facebook_friend_ids = FacebookService.from_user(user).facebook_friend_ids
+    add_friends(facebook_friend_ids, user)
+  end
+
+  def add_friends(facebook_friend_ids, user)
+    facebook_friend_ids.each do |facebook_id|
+      friend = User.find_by(facebook_id: facebook_id)
+      if friend
+        friend.follow(user) unless friend.following?(user)
+        user.follow(friend) unless user.following?(friend)
+      end
```

Comment:
```Ruby
# models/user.rb
def follow(other_user)
  active_relationships.create(following_id: other_user.id) 
end

def following?(other_user)
  following.include?(other_user)
end

# models/user_relationship
validates_uniqueness_of :follower_id, scope: :following_id
````

If I'm not mistaken, create will silently fail due to validation errors, so technically, the `unless` is not strictly necessary. However, i don't think it will save us a query. Instead of having it here, the validation itself will execute the same query.